### ISSUE TITLE
httpx 1.2.5 ignores -proxy when -unsafe used

### ISSUE THREAD

**Issue by geeknik** [projectdiscovery/httpx#913]
### httpx version:
1.2.5

### Current Behavior:
httpx 1.2.5 ignores `-proxy` when `-unsafe` is used.

### Expected Behavior:
httpx should respect `-proxy` regardless of `-unsafe` usage.

### Steps To Reproduce:
* `setup a proxy to list on 127.0.0.1:3128`
* `httpx -title -sc -td -unsafe -proxy http://127.0.0.1:3128 -u http://example.com`

**Comment by Mzack9999** [comment]
@geeknik http proxy is not yet supported by [rawhttp](https://github.com/projectdiscovery/rawhttp) as most proxies would reject non-rfc compliant requests. Anyway, the socks5 proxy should work fine. We can indeed attempt to implement a best-effort proxy use for rawhttp, but there is no guarantee that the proxy would reject the request. Do you have any particular use case for this functionality?

**Comment by geeknik** [comment]
>  Do you have any particular use case for this functionality?

Yes, but it's part of an upcoming blog post and I'd rather not publicly post the details until I can post the blog. 👍🏻  Feel free to follow-up on Discord if necessary. 😃

**Comment by Mzack9999** [comment]
Tracked at https://github.com/projectdiscovery/rawhttp/issues/85

**Comment by ehsandeep** [comment]
proxy is not supported with `unsafe` mode.
